### PR TITLE
Fix testProperty deprecation warning

### PR DESCRIPTION
**Pull request by newhoggy**

### CODE REVIEW

**Comment by newhoggy** [comment]
Unfortunately, this breaks compatibility with `tasty-hedgehog-1.1.0.0` hence the removal of `lts-8` and `lts-9`

I tried to make a version that is compatible with `CPP`, but unfortunately `CPP` pass doesn’t happen after the `-pgmF` preprocessor stage.

**Comment by newhoggy** [comment]
One possible way to fix this is to also require `tasty-discover` to be a library and include a compatibility shim in there.

File: library/Test/Tasty/Discover/Internal/Generator.hs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+{-# LANGUAGE CPP #-}
+
```
Comment (endgame) [inline library/Test/Tasty/Discover/Internal/Generator.hs:2]: Possibly not needed?

**Review by endgame — COMMENTED** [review]
Looks promising. I think if you stick with the `-pgmF` route this is probably as good as it's gonna get, unless you ask the users to leave information for `tasty-discover` in their `Main.hs` files.

**Comment by newhoggy** [comment]
In the interest of maintaining compatibility as much as possible, this PR will be replaced by https://github.com/haskell-works/tasty-discover/pull/32

**Pull request closed without merging**